core/drivers/crypto/stm32/stm32_cryp.c

```diff
@@ -1266,10 +1313,7 @@ static TEE_Result stm32_cryp_probe(const void *fdt, int node,
 	if (clk_enable(cryp_pdata.clock))
 		panic();
 
-	if (rstctrl && rstctrl_assert_to(cryp_pdata.reset, TIMEOUT_US_1MS))
-		panic();
-
-	if (rstctrl && rstctrl_deassert_to(cryp_pdata.reset, TIMEOUT_US_1MS))
+	if (stm32_cryp_reset())
```

Comment:
This should be part of a separate commit since it also introduces a delay, so it's not just code refactoring.

Reply:
Ok, I will split the commit in 2 and push another version with 3 commits.